### DISCUSSION TITLE
Extracting Chinese some texts are not right.

### DISCUSSION THREAD

**Discussion by buptyyf** [pymupdf/PyMuPDF#2367] [Q&A] [answered]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
In one pdf, pymupdf extracts some chinese character not right.

Example PDF：
[example.pdf](https://github.com/pymupdf/PyMuPDF/files/11302963/default.pdf)

code
```
doc = fitz.open(pdf_file_path)
doc[1].get_text()
```
<img width="1089" alt="image" src="https://user-images.githubusercontent.com/13342539/233823783-05d2797d-480c-4da2-a870-65ce68d77e8d.png">
<img width="682" alt="image" src="https://user-images.githubusercontent.com/13342539/233823798-80ee39e4-87f8-42d7-b59e-e31e98170ccf.png">

Many characters are not right.

How can I recognize and fix it?

**Comment by JorjMcKie** [comment]
This is a "Discussions" item, so I will first covert it.

**Comment by JorjMcKie** [comment]
This problem is caused by the fonts themselves. A text extraction software is dependent on the font's back-translation information, which is contained in the `/ToUnicode` table or directly given as a standard character encoding scheme, e.g. `/WinAnsiEncoding`.
Any wrong information there cannot be detected as such.
The only thing you can try is using OCR, sorry.

**Reply by buptyyf** [reply]
So,  I think it's no solution. I have no way to detect this problem, so that I can use OCR to solve it.

**Reply by buptyyf** [reply]
I have another question. Why pdf reader in mac or chrome can read this pdf correctly?

**Reply by JorjMcKie** [reply, marked as answer]
> I have another question. Why pdf reader in mac or chrome can read this pdf correctly?

**_Showing_** this PDF is not the problem. You talked about **_text extraction_**. If you create a page pixmap with PyMuPDF, you will get the right picture.
**But** if using e.g. Adobe Acrobat or any other PDF viewer and then selecting the text with the cursor you will get the same wrong result.
